### DISCUSSION TITLE
How to customize Marker

### DISCUSSION THREAD

**Discussion by andyZou0814** [patrykandpatrick/vico#443] [Questions] [answered]
I want to customize the content of `Marker` and display the text and icon I want to display, like the following.How can I do it?
thanks

x axia data: xxx
{icon1} y axia data: yyy

**Comment by Gowsky** [comment, marked as answer]
Hello @andyZou0814, The [`MarkerComponent`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.component.marker/-marker-component/?query=open%20class%20MarkerComponent(val%20label:%20TextComponent,%20val%20indicator:%20Component?,%20val%20guideline:%20LineComponent?)%20:%20Marker) allows for setting a custom [`MarkerLabelFormatter`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.marker/-marker-label-formatter/). In [`MarkerComponent#getLabel`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.marker/-marker-label-formatter/#-564124094%2FFunctions%2F186187275) you can return a [`Spannable`](https://developer.android.com/reference/kotlin/android/text/Spannable?hl=en) with a [`DynamicDrawableSpan`](https://developer.android.com/reference/android/text/style/DynamicDrawableSpan) that will display your icon.
You may also implement your custom [`Marker`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.marker/-marker/?query=interface%20Marker%20:%20ChartInsetter) implementation and draw whatever you wish straight onto a `Canvas`.

**Reply by andyZou0814** [reply]
Hello @Gowsky 
Thank you very much for your reply, I will try it
